Revision of CREATE TABLE doc

jseldess opened the pull request:
Copy edits to parameter definitions and other sections. Structural and copy edits to examples.

Also includes improvements to docs for `CREATE DATABASE`, `RENAME DATABASE`, and Indexes.

Fixes #312 

@dt or @RaduBerinde, mind taking a quick look?

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="35" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/docs/322)
<!-- Reviewable:end -->

uptimeDBA commented:
There's a note on the data-definition page about check constraints not being ready for prime time and that they are not validated by update statements. Looking at last weeks release notes I think this has been fixed [#6753](https://github.com/cockroachdb/cockroach/pull/6753) so maybe the note needs adjusting or removing? @dt @jseldess

dt commented on create-table.md:
```diff
@@ -4,87 +4,89 @@ toc: false
-If the table does **not** have a `PRIMARY KEY` constraint defined, a mandatory column called `rowid` of type `INT` will be added and automatically populated with a unique id that will be used as the primary key. The column will be added even if the table has an `UNIQUE` constraint.
+### Create a Table With a Primary Key
+
+In this example, we create a table with three columns. One column is given the [`PRIMARY KEY`](data-definition.html#primary-key) constraint, another is given the [`UNIQUE`](data-definition.html#unique) constraint, and the third has no constraints. Indexes are automatically created for columns with the `PRIMARY KEY` and `UNIQUE` constraints.
```
I still feel weird calling `PRIMARY KEY` a "constraint"... I guess pg does too though so maybe I just need get over it.

dt replied:
I think "Indexes are automatically created for columns with the `PRIMARY KEY`" is a little bit misleading -- an index isn't *created* for `PRIMARY KEY` really, and wording it this way may reinforce the (erroneous) idea that omitting it somehow saves on space or indexing costs.

Maybe something like "as a result of how CockroachDB stores rows, the  column(s) of the PRIMARY KEY are automatically indexed" or something to that effect might avoid the confusion?

dt commented on create-table.md:
```diff
@@ -4,87 +4,89 @@ toc: false
+
+### Create a Table Without a Primary Key
+
+In this example, we create a table with two columns, neither of which are given the the [`PRIMARY KEY`](data-definition.html#primary-key) constraint. As a result, a column called `rowid` of the type `INT` is added automatically as the primary key, with the `unique_row(id)` function used to ensure that new rows always default to unique `rowid` values. Also, an index is automatically created for `r ...
```
I might add something to basically saying that a PRIMARY KEY is _required_ because of how we store rows, which is why we add the extra field (which should also hopefully suggest to the reader that they should really be specifying their own).

RaduBerinde commented:
Review status: 0 of 4 files reviewed at latest revision, 3 unresolved discussions.

---

*[create-table.md, line 108 \[r10\]](https://reviewable.io:443/reviews/cockroachdb/docs/322#-KIh1qi2b1vjnKIwvfO6:-KIh1qi2b1vjnKIwvfO7:1929204100) ([raw file](https://github.com/cockroachdb/docs/blob/4d1ac49455f6f92f62a662321ff2c3975d82e75b/create-table.md#L108)):*
> ```Markdown
>   INDEX supp_id_prod_status_idx (supplier_id, product_status)
> );
> CREATE TABLE
> ```

Is this a stray line?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/322)*
<!-- Sent from Reviewable.io -->

jseldess commented:
Review status: 0 of 4 files reviewed at latest revision, 3 unresolved discussions.

---

*[create-table.md, line 34 \[r10\]](https://reviewable.io:443/reviews/cockroachdb/docs/322#-KIfADluV6OvIZeY5Cc2-r10-34:-KIhsDfSVphrBMcYOMzl:428400727) ([raw file](https://github.com/cockroachdb/docs/blob/4d1ac49455f6f92f62a662321ff2c3975d82e75b/create-table.md#L34)):*
<details><summary><i>Previously, dt (David Taylor) wrote…</i></summary>
> I think "Indexes are automatically created for columns with the `PRIMARY KEY`" is a little bit misleading -- an index isn't *created* for `PRIMARY KEY` really, and wording it this way may reinforce the (erroneous) idea that omitting it somehow saves on space or indexing costs.
> 
> 
> 
> Maybe something like "as a result of how CockroachDB stores rows, the  column(s) of the PRIMARY KEY are automatically indexed" or something to that effect might avoid the confusion?
</details>
Sorry, I remember you explaining this earlier. I've reframed this: We now state that every table requires a primary key. There's also a note about it not adding additional space. 

---

*[create-table.md, line 60 \[r10\]](https://reviewable.io:443/reviews/cockroachdb/docs/322#-KIfADluV6OvIZeY5Cc2-r10-60:-KIhsYxtxVJywnQMK5NV:481311790) ([raw file](https://github.com/cockroachdb/docs/blob/4d1ac49455f6f92f62a662321ff2c3975d82e75b/create-table.md#L60)):*
<details><summary><i>Previously, dt (David Taylor) wrote…</i></summary>
> I might add something to basically saying that a PRIMARY KEY is _required_ because of how we store rows, which is why we add the extra field when it is missing (which should also hopefully suggest to the reader that they should really be specifying their own).
</details>
Changed as described above. Name was misleading, too, so changed to "Create a Table (No Primary Key Defined)".

---

*[create-table.md, line 108 \[r10\]](https://reviewable.io:443/reviews/cockroachdb/docs/322#-KIh1qi2b1vjnKIwvfO6:-KIhsnBcpE6N_M0T7eh0:-467426180) ([raw file](https://github.com/cockroachdb/docs/blob/4d1ac49455f6f92f62a662321ff2c3975d82e75b/create-table.md#L108)):*
<details><summary><i>Previously, RaduBerinde wrote…</i></summary>
> Is this a stray line?
</details>
Not there anymore. I think you must've been looking at an earlier commit. I have to get better at squashing those. Sorry about that.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/322)*
<!-- Sent from Reviewable.io -->

RaduBerinde commented:
Review status: 0 of 4 files reviewed at latest revision, 3 unresolved discussions.

---

*[create-table.md, line 108 \[r10\]](https://reviewable.io:443/reviews/cockroachdb/docs/322#-KIh1qi2b1vjnKIwvfO6:-KIhyDqyR9-knAcw2Oka:831363541) ([raw file](https://github.com/cockroachdb/docs/blob/4d1ac49455f6f92f62a662321ff2c3975d82e75b/create-table.md#L108)):*
<details><summary><i>Previously, jseldess wrote…</i></summary>
> Not there anymore. I think you must've been looking at an earlier commit. I have to get better at squashing those. Sorry about that.
</details>
Ah no I get it now, this `CREATE TABLE` is the "output" of the command. It's a bit confusing, it would be great if we could somehow visually distinguish from what we send and what is received.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/322)*
<!-- Sent from Reviewable.io -->

jseldess commented:
*[create-table.md, line 108 \[r10\]](https://reviewable.io:443/reviews/cockroachdb/docs/322#-KIh1qi2b1vjnKIwvfO6:-KIiGBEliVCgUSw-dLSX:-361631535) ([raw file](https://github.com/cockroachdb/docs/blob/4d1ac49455f6f92f62a662321ff2c3975d82e75b/create-table.md#L108)):*
<details><summary><i>Previously, RaduBerinde wrote…</i></summary>
> Ah no I get it now, this `CREATE TABLE` is the "output" of the command. It's a bit confusing, it would be great if we could somehow visually distinguish from what we send and what is received.
</details>
Yeah, I agree that's confusing. I've used `>` to indicate a command. Hopefully that helps.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/322)*
<!-- Sent from Reviewable.io -->

jseldess commented:
Thanks, Paul. I removed that warning and added another stating that you shouldn't rename a column that has a check constraint.

<details><summary><i>Previously, uptimeDBA (Paul Steffensen) wrote…</i></summary>
> There's a note on the data-definition page about check constraints not being ready for prime time and that they are not validated by update statements. Looking at last weeks release notes I think this has been fixed [#6753](https://github.com/cockroachdb/cockroach/pull/6753) so maybe the note needs adjusting or removing? @dt @jseldess 
</details>


---

Review status: 0 of 4 files reviewed at latest revision, 3 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/322#-:-KIiN3iw9cgeJ7_56Npa:1479470644)*
<!-- Sent from Reviewable.io -->

dt commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

<details><summary><i>Previously, jseldess wrote…</i></summary>
> Thanks, Paul. I removed that warning and added another stating that you shouldn't rename a column that has a check constraint.
</details>


---

Review status: 0 of 6 files reviewed at latest revision, 4 unresolved discussions.

---

*[create-table.md, line 34 \[r10\]](https://reviewable.io:443/reviews/cockroachdb/docs/322#-KIfADluV6OvIZeY5Cc2-r10-34:-KIiOmH2weKwOkGwqkrQ:262468467) ([raw file](https://github.com/cockroachdb/docs/blob/4d1ac49455f6f92f62a662321ff2c3975d82e75b/create-table.md#L34)):*
<details><summary><i>Previously, jseldess wrote…</i></summary>
> Sorry, I remember you explaining this earlier. I've reframed this: We now state that every table requires a primary key. There's also a note about it not adding additional space. 
</details>
:thumbsup: 

---

*[data-definition.md, line 229 \[r12\]](https://reviewable.io:443/reviews/cockroachdb/docs/322#-KIiOtzFdBp8LVVvXpNq:-KIiOtzFdBp8LVVvXpNr:-1193260894) ([raw file](https://github.com/cockroachdb/docs/blob/93705d99a2fa9e302ea8de1d87e78ea880890274/data-definition.md#L229)):*
> ```Markdown
> ## Indexes
> 
> Indexes are used to quickly locate data without having to look through every row of a table. They can be created during table creation ([`CREATE TABLE`](create-table.html)) or separately using the [`CREATE INDEX`](create-index.html) statement. During table creation, indexes are automatically created for the primary key of a table and any columns with a unique constraint.
> ```

nit: this contradicts the above where we say primary doesn't actually *create* an additional index, but it's probably ok.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/322#-:-KIiOkwHMsggHYe7ijAA:1417088721)*
<!-- Sent from Reviewable.io -->

jseldess commented:
*[data-definition.md, line 229 \[r12\]](https://reviewable.io:443/reviews/cockroachdb/docs/322#-KIiOtzFdBp8LVVvXpNq:-KIiQxAfmghaHp97sjN-:-1571636294) ([raw file](https://github.com/cockroachdb/docs/blob/93705d99a2fa9e302ea8de1d87e78ea880890274/data-definition.md#L229)):*
<details><summary><i>Previously, dt (David Taylor) wrote…</i></summary>
> nit: this contradicts the above where we say primary doesn't actually *create* an additional index, but it's probably ok.
</details>
Tweaked a bit.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/322)*
<!-- Sent from Reviewable.io -->
